param/parameterized.py

```diff
@@ -1488,21 +1488,28 @@ def trigger(self_, *param_names):
         """
         Trigger watchers for the given set of parameter names. Watchers
         will be triggered whether or not the parameter values have
-        actually changed.
+        actually changed. As a special case, the value will actually be
+        changed for a Parameter of type Event, setting it to True so
+        that it is clear which Event parameter has been triggered.
         """
+        trigger_params = [p for p in self_.self_or_cls.param
+                          if (self_.self_or_cls.param[p].__class__.__name__ == 'Event')]
+        for pname in [p for p in trigger_params if p in param_names]:
+            setattr(self_.self_or_cls, pname, True)
```

Comment:
If you're only triggering a trigger param shouldn't it return after this section? I guess an empty `set_param` is a no-op, still seems safer though. Also shouldn't this use `set_param` rather than `setattr` so the events are batched?

Reply:
Good point about using `set_param`, should be an easy fix. As for returning, I don't think that works as not all parameters being triggered are necessarily `Events`...unless you mean returning if `set(trigger_params) == set(param_names)` leaving an empty list for the rest of the method to process (which is fine).

Reply:
Yes that's what I mean.

Reply:
Addressed in 125c222